generator/src/generator/error_events.rs

```diff
@@ -490,6 +490,42 @@ fn generate_events(out: &mut Output, module: &xcbgen::defs::Module) {
         outln!(out, "}}");
         outln!(out, "");
 
+        outln!(
+            out,
+            "/// Get the additional length if any (in 4 byte units) of this X11 event.",
+        );
+        outln!(out, "pub fn length(&self) -> Option<u32> {{");
```

Comment:
I'm not quite sure what use you have in mind for this function, but why not return `u32` without the `Option`? Basically `old_result.unwrap_or(0)`.